Popup/Popup.js

```diff
@@ -86,20 +166,21 @@ const PopupBase = kind({
 	},
 	styles: {
 		css: componentCss,
-		className: 'popup'
+		className: 'popup',
+		publicClassNames: ['popup', 'body', 'popupTransitionContainer', 'top', 'right', 'bottom', 'left']
 	},
 	computed: {
-		className: ({centered, closeButton, position, styler, title}) => styler.append({top: position === 'top', withCloseButton: closeButton, withTitle: title, centered}),
+		className: ({centered, closeButton, position, styler, title}) => styler.append(position, {withCloseButton: closeButton, withTitle: title, centered}),
 		transitionType : ({position}) => position === 'center' ? 'fade' : 'slide',
-		direction : ({position}) => position === 'center' ? 'down' : 'up'
+		direction : ({position}) => transitionDirection[position]
 	},
-	render: ({buttons, children, closeButton, css, direction, noAnimation, onClose, onHide, open, skin, title, transitionType, ...rest}) => {
+	render: ({buttons, children, closeButton, css, direction, noAnimation, onClose, onHide, open, position, skin, title, transitionType, ...rest}) => {
 		const wideLayout = (skin === 'carbon');
 		delete rest.centered;
 
 		return (
 			<TransitionContainer
-				noAnimation={noAnimation}
+				noAnimation={position === 'fullscreen' ? true : noAnimation}
```

Comment:
Why this line needed? Is this from the requirement?

Reply:
No, it was not. I corrected it. Thank you.